Packs/QuestKace/Integrations/QuestKace/QuestKace.py

```diff
@@ -0,0 +1,785 @@
+            raise DemistoException("Could not get cookie")
+        return token, cookie
+
+    def update_token(self):
+        """Update cookie and token.
+            Returns:
+                Tuple of token and cookie.
+        """
+        self._token, self._cookie = self.get_token()
+
+    def get_cookie(self, res_cookie: dict) -> str:
+        """Get a cookie from an cookie object in the needed format for the requests.
+            Args:
+                res_cookie: part of the response that the cookie is inside it.
+            Returns:
+                string that will be sent in the requests which represents the cookie in the header.
+        """
+        KACE_CSRF_TOKEN = res_cookie.get("KACE_CSRF_TOKEN").__dict__.get('value')
+        x_dell_auth_jwt = res_cookie.get("x-dell-auth-jwt").__dict__.get('value')
+        kboxid = res_cookie.get("kboxid").__dict__.get('value')
+        KACE_LAST_USER_SECURE = res_cookie.get("KACE_LAST_USER_SECURE").__dict__.get('value')
+        KACE_LAST_ORG_SECURE = res_cookie.get("KACE_LAST_ORG_SECURE").__dict__.get('value')
+
+        cookie = f'KACE_LAST_USER_SECURE={KACE_LAST_USER_SECURE}; KACE_LAST_ORG_SECURE={KACE_LAST_ORG_SECURE};' \
+                 f' kboxid={kboxid}; x-dell-auth-jwt={x_dell_auth_jwt}; KACE_CSRF_TOKEN={KACE_CSRF_TOKEN}'
+        return cookie
+
+    def token_request(self, url: str, headers: Optional[dict] = None, data: Optional[str] = None) -> dict:
+        """login request for initiating a connection with the product.
+        Args:
+            url: full url that the request will be sent to.
+            headers: headers of the request.
+            data: data of the request which includes username and password.
+        Returns:
+            Dictionary of the response from the product.
+        """
+        try:
+            response = requests.request("POST", url, headers=headers, data=data, verify=self._verify)
+        except requests.exceptions.SSLError:
+            err_msg = 'SSL Certificate Verification Failed - try selecting \'Trust any certificate\' checkbox in' \
+                      ' the integration configuration.'
+            raise DemistoException(err_msg)
+        except requests.exceptions.ConnectionError:
+            raise DemistoException("Invalid url , Failed to establish a connection")
+        if response.status_code == 401:
+            raise DemistoException("Error Code 401 - Invalid user or password")
+        return response.__dict__
+
+    def machines_list_request(self, filter_fields: Optional[str] = None):
+        """List of machines.
+           Returns:
+               Response from API.
+        """
+        headers = {
+            'Accept': 'application/json',
+            'x-dell-csrf-token': self._token,
+            'x-dell-api-version': '5',
+            'Cookie': self._cookie
+        }
+        if filter_fields:
+            return self._http_request("GET", url_suffix=f"/inventory/machines?filtering={filter_fields}",
```

Comment:
let's build the `url_suffix` accordingly to the `filter_fields`
and use only 1 return in the function